lib/common/strings.c

```diff
@@ -872,14 +872,30 @@ pcmk__parse_ll_range(const char *srcstring, long long *start, long long *end)
  * Search \p lst for \p s, taking case into account.  As a special case,
  * if "*" is the only element of \p lst, the search is successful.
  *
- * \param[in]  lst  List to search
- * \param[in]  s    String to search for
+ * Behavior can be changed with various flags:
+ *
+ * - pcmk__str_casei - By default, comparisons are done taking case into
+ *                     account.  This flag makes comparisons case-insensitive.
+ * - pcmk__str_null_matches - If the input string is NULL, return TRUE.
+ *
+ * \note The special "*" matching rule takes precedence over flags.  In
+ *       particular, "*" will match a NULL input string even without
+ *       pcmk__str_null_matches being specified.
```

Comment:
Not a suggestion for anytime soon, but I wonder if pcmk__str_star_matches would make sense :)